python/ray/data/datasource/file_based_datasource.py

```diff
@@ -292,6 +297,16 @@ def _expand_directory(path: str,
     return zip(*sorted(filtered_paths, key=lambda x: x[0]))
 
 
+def _is_url(path) -> bool:
+    return isinstance(path, str) and "://" in path
```

Comment:
I feel this is kind of sketchy. Is this actually a bug in the pyarrow filesystem?

Reply:
Or should the user be encoding it themselves on the read end?

Reply:
Yes it is, and this is basically the same check that Pandas uses to check whether they're getting `fsspec`-compatible URLs, so I wouldn't worry about this being sketchy. https://github.com/pandas-dev/pandas/blob/c5e236b6cc586a627f9364843ce00d2f293d57b4/pandas/io/common.py#L215-L224

Reply:
>Or should the user be encoding it themselves on the read end?

If we're aiming to be more user-friendly than Arrow, e.g. closer to Pandas UX, then this is a bug.

Reply:
Ok--- so basically this is a workaround for a pyarrow Filesystem bug. Can we make a note of this?

Reply:
^ can you throw that link in a comment? 

Also why not `try: assert urllib.parse(url).scheme`?

Reply:
@wuisawesome I was hoping to closely mirror Pandas since they have the desired UX, but agreed, I like using urllib's parsing more. I'll add that.